test/utils/setup.ts

```diff
@@ -1,14 +1,50 @@
 /*
     Setup -- start dynamodb instance
  */
+import { spawn } from 'child_process'
+const waitPort = require('wait-port')
 import DynamoDbLocal from 'dynamo-db-local'
 
 const PORT = 4567
```

Comment:
```suggestion
const PORT = parseInt(process.env.DYNAMODB_PORT || 4567)
```

Reply:
Could we get to just one env var?  i.e

DYNAMO_DOCKER_PORT?

Reply:
Could we get rid of the DOCKER=true and trigger off the DYNAMODB_PORT?

Reply:
I think I understand, does this make sense:

![docker-running-tests-v2](https://user-images.githubusercontent.com/30017294/129682099-cab87c58-e926-4d4a-be7f-ebeeddc6defc.gif)

This would mean that we can have long-running local dynamodb, without worrying about port collision when running tests, is that the idea?

Reply:
Yes. You may need to do the PORT references on line 8 and 45?

The only other possibility to consider would be to call it DYNAMODB_DOCKER and you could set that to a port or true. If true, then use an internal pre-defined port. Otherwise, it gives the port number.

That way we can document for most people to use DYNAMODB_DOCKER=true but others can do DYNAMODB_DOCKER=17070 or similar.

What I wanted was to default to one env var (PID excepted) and be as simple as possible.

Thanks for your work on this. Great to have your help.

Reply:
Ok, I think I have some comments on this how we might go forward to make this straightforward, so, please let me know what you think.

1. User can specify PORT that local dynamodb runs on
2. User can specify DOCKER as a truthy value 'yes', 'true', '1'

## Examples

```bash
npm test 
> port = 4567, does not use docker
PORT=1234 npm test
> port = 1234, does not use docker
DOCKER=true npm test
> port = 4567, uses docker
DOCKER=true PORT=1234 npm test
> port = 1234, uses docker
```

In this way, we separate concerns:

- Do you want docker (yes/no)
- Do you want local dynamoDB on a specific port (provide a port number)

## Reason

The user has the same experience, regardless of docker or not, they just get a port that points at local docker.

Let me know what you think :thinking:

Reply:
Sounds good. Sorry to give you a bad steer on this.

Reply:
Let me know when you think this is complete.

Reply:
Happy with this one, going to resolve :+1: